.tekton/tasks/rebase-pr.yaml

```diff
@@ -0,0 +1,124 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: rebase-pr
+spec:
+  description: |
+    rebase the pull request.
+  params:
+    - name: pull_request_number
+    - name: TARGET_GH_OWNER
+      description: Owner of GitHub repository of the infra-deployments code
+      default: redhat-appstudio
+    - name: TARGET_GH_REPO
+      description: GitHub repository of the infra-deployments code
+      default: infra-deployments
+    - name: source_branch
+  results:
+    - name: source-current-revision
+  workspaces:
+   - name: shared-file-path
+  steps:
+    - name: rebase-pull-request
+      image: quay.io/redhat-appstudio/github-app-token@sha256:b4f2af12e9beea68055995ccdbdb86cfe1be97688c618117e5da2243dc1da18e
+      workingDir: $(workspaces.shared-file-path.path)
+      env:
+        - name: GITHUB_GRAPHQL_API_URL
+          value: https://api.github.com/graphql
+        - name: PULL_REQUEST_NUMBER
+          value: $(params.pull_request_number)
+        - name: SOURCE_BRANCH
+          value: $(params.source_branch)
+        - name: TARGET_GH_OWNER
+          value: "$(params.TARGET_GH_OWNER)"
+        - name: TARGET_GH_REPO
+          value: "$(params.TARGET_GH_REPO)"
+      script: |
```

Comment:
I'm a bit surprised you need something as complicated as this.

I would have expected a simple script such as:
```
git fetch
git rebase origin/main
```

Reply:
I already tried this out but not sure somewhere it stuck out when I do rebase like this. Note with this I had also exported `GITHUB_TOKEN` env variable so that it won't fail on authentication

Reply:
Done. Working with git command now